### PR TITLE
Eliza Stuff

### PR DESCRIPTION
**Pull request by leogrim**
@ryanpbrewster @andrewconner 
- Eliza APIs work with ExternalId[MessageThread] or PublicId[Keep], 
- comment threads are interned
- deprecate ExternalId[Message] in favor of PublicId[Message]

### CODE REVIEW

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/NotificationMessagingCommander.scala [hunk 1]
```diff
@@ -45,14 +45,12 @@ class NotificationMessagingCommander @Inject() (
...
 
-  def sendUnreadNotificationsWith(notif: Notification, recipient: Recipient): Unit = {
+  private def sendUnreadNotificationsWith(notif: Notification, recipient: Recipient): Unit = {
     recipient match {
       case UserRecipient(user) =>
         val unreadMessages = messagingCommander.getUnreadUnmutedThreadCount(user)
         val unreadNotifications = notificationCommander.getUnreadNotificationsCount(Recipient(user))
         webSocketRouter.sendToUser(user, Json.arr("unread_notifications_count", unreadMessages + unreadNotifications, unreadMessages, unreadNotifications))
-        val pushNotif = MessageThreadPushNotification(ExternalId[MessageThread](notif.externalId.id), unreadMessages + unreadNotifications, None, None)
-        pushNotifier.notifyUser(user, pushNotif, false)
```
Comment (leogrim) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/NotificationMessagingCommander.scala:55]: @andrewconner pretty sure this didn't make sense, was called when notifs were marked unread, don't see why we'd send a push notif.

Comment (andrewconner) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/NotificationMessagingCommander.scala:55]: Perhaps to update other clients? Need to follow the logic, but that comes to mind

File: kifi-backend/modules/eliza/app/com/keepit/eliza/model/MessageThread.scala [hunk 2]
```diff
@@ -168,6 +170,27 @@ object MessageThread {
...
+  def toIdString(id: MessageThreadId)(implicit publicIdConfiguration: PublicIdConfiguration): String = id match {
+    case ThreadExternalId(threadId) => threadId.id
+    case KeepId(keepId) => Keep.publicId(keepId).id
+  }
+
+  def fromIdString(idStr: String)(implicit publicIdConfiguration: PublicIdConfiguration): Option[MessageThreadId] = {
+    ExternalId.asOpt[MessageThread](idStr).map(ThreadExternalId(_)) orElse Keep.validatePublicId(idStr).flatMap(pubId => Keep.decodePublicId(pubId).map(KeepId(_)).toOption)
+  }
+
+  implicit def format(implicit publicIdConfiguration: PublicIdConfiguration) = Format[MessageThreadId](
```
Comment (ryanpbrewster) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/model/MessageThread.scala:186]: This is a pretty weird thing. The `Format[T]` actually reads in a `PublicId[T]` _and_ tries to convert it to an `Id[T]`? I guess unavoidable.

File: kifi-backend/modules/eliza/app/com/keepit/eliza/model/MessageThread.scala [hunk 3]
```diff
@@ -168,6 +170,27 @@ object MessageThread {
...
+case class ThreadExternalId(threadId: ExternalId[MessageThread]) extends MessageThreadId
+case class KeepId(keepId: Id[Keep]) extends MessageThreadId
+object MessageThreadId {
+  def toIdString(id: MessageThreadId)(implicit publicIdConfiguration: PublicIdConfiguration): String = id match {
+    case ThreadExternalId(threadId) => threadId.id
+    case KeepId(keepId) => Keep.publicId(keepId).id
+  }
+
+  def fromIdString(idStr: String)(implicit publicIdConfiguration: PublicIdConfiguration): Option[MessageThreadId] = {
+    ExternalId.asOpt[MessageThread](idStr).map(ThreadExternalId(_)) orElse Keep.validatePublicId(idStr).flatMap(pubId => Keep.decodePublicId(pubId).map(KeepId(_)).toOption)
```
Comment (andrewconner) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/model/MessageThread.scala:183]: `decodePublicId` fails fast if it has the wrong prefix, so you can safely just jump straight to using it

File: kifi-backend/modules/eliza/app/com/keepit/eliza/model/MessageThread.scala (hunk 2 above)
Comment (leogrim) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/model/MessageThread.scala:186]: Well, the sooner with have `keepId` everywhere, the sooner we can get rid of this.

**Pull request merged**